src/Microsoft.AspNetCore.Authentication.Impl/AuthenticationServiceCollectionExtensions.cs

```diff
@@ -46,8 +45,7 @@ public static IServiceCollection AddAuthentication(this IServiceCollection servi
             return services;
         }
 
-        // REVIEW: rename to just AddScheme?
-        public static IServiceCollection AddSchemeHandler<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions)
+        public static IServiceCollection AddScheme<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions, bool canHandleRequests)
```

Comment:
By scheme, I mean scheme handler. In your case, you'd have only one scheme handler (WS-Fed), with 2K different sets of options, resolved dynamically.